src/lumen/hardware/sd_card.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "nlohmann/json.hpp"
+
+#include <string>
+
+namespace lumen::hardware {
+
+/// Initialize the SD card.
+void init_sd_card();
+
+/** Read a file stream into a JSON object.
+ *
+ * \param filepath The path to the file to read from.
+ *
+ * \returns The file as a JSON object.
+ *
+ * \note `filepath` must be prepended by '/'
+ */
+nlohmann::json read_json(std::string filepath);
+
+/** Write a JSON object to a file.
+ *
+ * \param filepath The path to the file to write to.
+ *
+ * \param data The JSON object to write.
+ *
+ * \note `filepath` must be prepended by '/'
+ */
+void write_json(std::string filepath, nlohmann::json data);
```

Comment:
This should also be a const&

Reply:
I believe that the `json` data will be optimized. You can refer to the use case in `Context.store_activity()`.